halo2_proofs/src/poly/commitment.rs

```diff
@@ -81,18 +81,19 @@ impl<C: CurveAffine> Params<C> {
             g
         };
 
-        // Let's evaluate all of the Lagrange basis polynomials
-        // using an inverse FFT.
-        let mut alpha_inv = E::Scalar::ROOT_OF_UNITY_INV;
+        let mut g_lagrange_projective = vec![E::G1::group_zero(); n as usize];
+        let mut root = E::Scalar::ROOT_OF_UNITY_INV.invert().unwrap();
         for _ in k..E::Scalar::S {
-            alpha_inv = alpha_inv.square();
+            root = root.square();
         }
-        let mut g_lagrange_projective = g_projective;
-        best_fft(&mut g_lagrange_projective, alpha_inv, k);
-        let minv = E::Scalar::TWO_INV.pow_vartime(&[k as u64, 0, 0, 0]);
-        parallelize(&mut g_lagrange_projective, |g, _| {
-            for g in g.iter_mut() {
-                *g *= minv;
+        let multiplier =
+            (s.pow_vartime(&[n as u64]) - E::Scalar::one()) * E::Scalar::from(n).invert().unwrap();
```

Comment:
Just unsure about the fact that depending on the `n` you provide, it might be NQR (Non-Quadratic-Residue) or not. And that directly makes the `E::Scalar::from(n).invert().unwrap()` fail.

Reply:
n here is 1<<k, usually 1<<12 to 1<<28. Will these values panic?

Reply:
I would say no.
Anyway, we can hint a bit better the error using `expect` and providing a small explanation like: `Scalar inverse not found during setup` or similar.